### ISSUE TITLE
detect redefinition of table key

### ISSUE THREAD

**Issue by johnfoconnor** [mpeterv/luacheck#76]
would be nice to detect if a key is defined more than once in a static table declaration

```
local colors = {
    RED = 1,
    BLUE = 2,
    YELLOW = 3,
    RED = 4,
}
```

in this case it would be good for it to detect that the key `RED` is redefined within the table declaration

**Comment by mpeterv** [comment]
This is detected since 0.14.0:

```
luacheck - <<< 'local colors = {
     RED = 1,
     BLUE = 2,
     YELLOW = 3,
     RED = 4,
 }'

```
Checking stdin                                    2 warnings

    stdin:1:7: unused variable colors
    stdin:2:6: value assigned to field RED is unused

Total: 2 warnings / 0 errors in 1 file
```

It's reported as an unused value of the first key rather than a redefinition because in general redefining keys with several assignments is fine as long as the value is used in between.

**Comment by johnfoconnor** [comment]
ah sorry i was just reading through the docs at the warning error codes. Looks like this is covered by W314 "Value of a field in a table literal is unused."?

I interpreted that as no one is accessing that value programatically, not that the value is replaced in the table literal definition

**Comment by mpeterv** [comment]
> I interpreted that as no one is accessing that value programatically, not that the value is replaced in the table literal definition

Yes, the former is sort of subset of the latter. If generic unused field detection will ever be implemented (relatively difficult to do in a flow-sensitive way) it will use the same warning.
